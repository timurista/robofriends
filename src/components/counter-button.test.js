import React from 'react';
import { shallow, render } from 'enzyme'
import CounterButton from './counter-button'


describe('CounterButton component', () => {
  const intialProps = {}
  const setup = (props: {}) => {
    const newProps = {...intialProps, ...props};
    return shallow(<CounterButton {...newProps}/>)
  }

  it('CounterButton component exists', () => {
    const wrapper = setup();
    expect(wrapper.debug()).toMatchSnapshot()
  })

  it('CounterButton updates the color', () => {
    const wrapper = setup({ color: 'red'});
    expect(wrapper.props().color).toBe('red')
  })

  it('cliking button upodates counter', () => {
    const wrapper = setup();
    const counter = () => wrapper.find('[data-test-id="counter-button"]');
    counter().simulate('click');
    expect(counter().text()).toBe('Count: 1')
    counter().simulate('click');
    expect(counter().text()).toBe('Count: 2')
  })

  it('only changes state if count has changed', () => {
    const wrapper = setup();
    expect(wrapper
      .instance()
      .shouldComponentUpdate({ count: 0})
    ).toBe(false)
  })
})