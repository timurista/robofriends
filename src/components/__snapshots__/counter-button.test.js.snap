// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CounterButton component CounterButton component exists 1`] = `
"<button data-test-id=\\"counter-button\\" color={[undefined]} onClick={[Function]}>
  Count: 
  0
</button>"
`;
